tests/client_test.go

```diff
@@ -374,3 +374,24 @@ func (s *IntegrationTestSuite) TestShutdown() {
 
 	require.NoError(err)
 }
+
+func (s *IntegrationTestSuite) TestManyOps() {
+	require := require.New(s.T())
+	conn, err := s.remote.Dial(s.client)
+	require.NoError(err)
+	defer conn.Close()
+	var wg sync.WaitGroup
+	for i := 0; i < 2000; i++ {
```

Comment:
Can you make this 20000 instead? `maxConnPendingRequests` is `1024`, so `2000` isn't big enough to create a situation where we are waiting for a goroutine to become available. 

The situation this would help test would be behavior of the server when there are a significant number of requests that are waiting and there aren't any available goroutines to handle them, and if it can gracefully recover from that state without any large delays.

Reply:
It might be even better to do something like `i < s.server.maxConnPendingRequests * 20` so these aren't magic numbers, but then the value would have to be exported so maybe just a comment is fine.

Reply:
Unfortunately the race detector limits the number of goroutines we can have open at once. We'd also need to alert if there was a "big delay", and at that point we're looking at scheduler details.  This catches the most likely bug of not putting back the token.

Reply:
Ah didn't even think of the race detector issue, looks like it's a fairly low limit too about 8k. Looks good otherwise!